scipy/stats/stats.py

```diff
@@ -6212,31 +6212,61 @@ def _broadcast_concatenate(xs, axis):
     return res
 
 
-def _data_partitions(data, permutations, size_a, axis=-1, random_state=None):
-    """All partitions of data into sets of given lengths, ignoring order"""
+def _batch_generator(iterable, batch):
+    """A generator that yields batches of elements from an iterable"""
```

Comment:
To review this, it might be helpful to refer to [Working of For Loop Iterators](https://www.programiz.com/python-programming/iterator#for-loop). This is basically how a Python `for` loop works with a tiny bit of extra logic for batching.